Sources/ApolloCodegenLib/Templates/MockObjectTemplate.swift

```diff
@@ -56,21 +59,38 @@ struct MockObjectTemplate: TemplateRenderer {
       
       \(accessControlModifier(for: .parent))\
       extension Mock where O == \(objectName) {
-        \(conflictingFieldNameProperties(fields))
+        \(conflictingFieldNameProperties(fields.map { $0.0 }))
         convenience init(
           \(fields.map { """
-            \($0.propertyName)\(ifLet: $0.initializerParameterName, {" \($0)"}): \($0.mockType)? = nil
+            \($0.0.propertyName)\(ifLet: $0.0.initializerParameterName, {" \($0)"}): \($0.0.mockType)? = nil
             """ }, separator: ",\n")
         ) {
           self.init()
-          \(fields.map { "_set(\($0.initializerParameterName ?? $0.propertyName), for: \\.\($0.propertyName))" }, separator: "\n")
+          \(fields.map {
+            return "_set\(getMockSetType($0.1))(\($0.0.initializerParameterName ?? $0.0.propertyName), for: \\.\($0.0.propertyName))"
+          }, separator: "\n")
         }
       }
       """) : TemplateString(stringLiteral: "")
     )
     
     """
   }
+  
+  private func getMockSetType(_ graphQLType: GraphQLType) -> String {
```

Comment:
Not sure I like the name of this function, open to other suggestions here.

Reply:
Maybe just remove the `get` - `mockSetType` or something completely different in `mockFunctionDescriptor`.